File: src/main/webui/src/model/turret/Bullet.ts
import Influence from '../common/Influence';
import Modifiers from '../common/Modifiers';
import BulletSkin from './BulletSkin';

export default class Bullet {
    name: string;
    label: string;
    description: string;
    size: string;
    modifiers: Modifiers;
    availableSkins: BulletSkin[];
    compatibleChassis: string[];
    influence?: Influence;
    constructor(name: string, label: string, description: string, size: string, modifiers: Modifiers, 
        availableSkins: BulletSkin[], compatibleChassis: string[], influence?: Influence) {
        this.name = name;
        this.label = label;
        this.description = description;
        this.size = size;
        this.modifiers = modifiers;
        this.availableSkins = availableSkins;
        this.compatibleChassis = compatibleChassis;
        this.influence = influence;
    }
}

File: src/main/java/io/tbbc/cf/bullet/Bullet.java
package io.tbbc.cf.bullet;

import io.tbbc.cf.bullet.skin.BulletSkin;
import io.tbbc.cf.common.Size;
import io.tbbc.cf.modifier.Modifiers;
import io.tbbc.cf.turret.chassis.ChassisType;

import java.util.List;

public record Bullet(String name, String label, Size size, List<ChassisType> compatibleChassis,
                     Modifiers modifiers, List<BulletSkin> availableSkins) {
}
